src/PrestaShopBundle/Controller/Admin/Configure/AdvancedParameters/AdministrationController.php

```diff
@@ -162,7 +155,8 @@ protected function processForm(Request $request, FormHandlerInterface $formHandl
             try {
                 $formHandler->save($data);
             } catch (DataProviderException $e) {
-                $this->flashErrors($this->getErrorMessages($e->getInvalidConfigurationDataErrors()));
+                $errorMessageFactory = $this->get('form.invalid_configuration_error_message_factory');
```

Comment:
@matks Is it new behavior ?
Error Messages are not anymore in the controller ?

Reply:
What's important I think is not to have the error messages inside the Controller class, but to do the mapping between "catched Exception" and "error message" at the controller level. We want exceptions thrown in the logic below the controller to bubble up to the controller so that the controller can decide what it wants to do with it
- let the Exception un-catched
- catch the Exception and display an error message instead
- catch the Exception and silent it (user does not see it)

We want it to happen in the Controller so that module developers that would like to modify this behavior can do it by overriding/decorating part of the controller.

So IMO change is good because
- the try/catch is in the Controller
- the choice to display message `$this->flashErrors()` is in the Controller

Where we store the error messages... this is not dictated by our architecture 😊